charts/fission-all/templates/fluentbit/fluentbit.yaml

```diff
@@ -130,6 +130,7 @@ spec:
 {{- if .Values.logger.enableSecurityContext }}
           securityContext:
             privileged: true
+            runAsUser: 0
```

Comment:
We can you provider 

Probably we should modify main values so that user can their own options.
```
logger:
     enableSecurityContext: true/false
     securityContext: {}
     # Passing security context if enabled
     # securityContext:
     #      privilege: true
     #.     runAsUser: <privilege user>
````
 I would recommend not to use root user for security reasons.

Reply:
ok, will look at an alternative method

Reply:
@sanketsudake  I have changed the runAsUser to a chmod in the initContainer. This means that its not using a root user to run the container, but allows the non-root user to write into the /var/log/fission directory.  Hopefully this will be more acceptable. Ideally I would use a chown to the user who is running within the container but I am not sure how to do that.

Reply:
@markretallack As @sanketsudake has mentioned that we should provide user with options to configure `runAsUser` value in securityContext. We have created a patch with proper explanation for logger's securityContext field. Please try the patch.

```
diff --git a/charts/fission-all/templates/fluentbit/fluentbit.yaml b/charts/fission-all/templates/fluentbit/fluentbit.yaml
index f54a400f..9718fe97 100644
--- a/charts/fission-all/templates/fluentbit/fluentbit.yaml
+++ b/charts/fission-all/templates/fluentbit/fluentbit.yaml
@@ -128,10 +128,8 @@ spec:
             - name: docker-log
               mountPath: /var/lib/docker/containers
               readOnly: true
-{{- if .Values.logger.enableSecurityContext }}
           securityContext:
-            privileged: true
-{{- end }}
+            {{- toYaml .Values.logger.securityContext | nindent 12 }}
         - name: fluentbit
 {{- if .Values.repository }}
           image: "{{ .Values.logger.fluentdImageRepository }}/{{ .Values.logger.fluentdImage }}:{{ .Values.logger.fluentdImageTag }}"
diff --git a/charts/fission-all/values.yaml b/charts/fission-all/values.yaml
index f0b668f8..8225725f 100644
--- a/charts/fission-all/values.yaml
+++ b/charts/fission-all/values.yaml
@@ -620,11 +620,22 @@ logger:
 
   ## Fluent-bit writes/reads it’s own sqlite database to record a history of tracked
   ## files and a state of offsets, this is very useful to resume a state if the ser-
-  ## vice is restarted. For Kubernetes environment with constraints like OpenShift,
+  ## vice is restarted.
+  ##
+  ## For Kubernetes environment with constraints like OpenShift,
   ## the containers are limited to write hostPath volume. Hence, we have to enable
   ## security context and set privileged to true.
   ##
-  enableSecurityContext: false
+  ## The user ID in runAsUser should have access to the `/var/log` and
+  ## `/var/lib/docker/containers` directories on your host.
+  ## On many hosts, this user might be root ,i.e., `runAsUser: 0`.
+  ## Although it is recommended not to use root user for security reasons.
+  ##
+  ## The `/var/log` and `/var/lib/docker/containers` directories on host are mounted
+  ## to logger container with volumeType `HostPath`.
+  securityContext: {}
+    # privileged: true
+    # runAsUser: 0
 
   ## Enable PodSecurityPolicies to allow privileged container
   ## Only required in some clusters and when enableSecurityContext is true

```